sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/util/ActiveWindowSet.java

```diff
@@ -94,9 +95,10 @@ void onMerge(Collection<W> toBeMerged, Collection<W> activeToBeMerged, W mergeRe
   }
 
   /**
-   * Remove EPHEMERAL windows since we only need to know about them while processing new elements.
+   * Remove EPHEMERAL windows and remaining NEW windows since we only need to know about them
+   * while processing new elements.
    */
-  void removeEphemeralWindows();
+  void garbageCollect();
```

Comment:
`gcTemporaryWindows` or (tentative windows) or something like that? I don't want us to confuse this with the garbage collection of expired windows.